src/Config/Validator/Build/BalerSupport.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Config\Validator\Build;
+
+use Magento\MagentoCloud\Config\Magento\Shared\ReaderInterface;
+use Magento\MagentoCloud\Config\Validator;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+use Symfony\Component\Process\ExecutableFinder;
+
+/**
+ * Verify that the build configuration is ready for baler.
+ */
+class BalerSupport implements ValidatorInterface
+{
+    /**
+     * @var ExecutableFinder
+     */
+    private $finder;
+
+    /**
+     * @var Validator\ResultFactory
+     */
+    private $resultFactory;
+
+    /**
+     * @var ReaderInterface
+     */
+    private $configReader;
+
+    /**
+     * @param Validator\ResultFactory $resultFactory
+     * @param ExecutableFinder $finder
+     * @param ReaderInterface $configReader
+     */
+    public function __construct(
+        Validator\ResultFactory $resultFactory,
+        ExecutableFinder $finder,
+        ReaderInterface $configReader
+    ) {
+        $this->resultFactory = $resultFactory;
+        $this->finder = $finder;
+        $this->configReader = $configReader;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        $errors = [];
+
+        if ($this->finder->find('baler') === null) {
+            $errors[] = 'Path to baler executable could not be found.'
+                . ' The Node package may not be installed or may not be linked.';
+        }
+
+        $magentoConfig = $this->configReader->read();
+
+        if (!isset($magentoConfig['modules']['Magento_Baler']) || !$magentoConfig['modules']['Magento_Baler']) {
+            $errors[] = 'The Magento_Baler module is not installed or is disabled.';
+        }
+
+        $jsConfig = $magentoConfig['system']['default']['dev']['js'] ?? [];
+
+        if (!isset($jsConfig['enable_baler_js_bundling']) || !$jsConfig['enable_baler_js_bundling']) {
+            $errors[] = 'The Magento config dev/js/enable_baler_js_bundling must be enabled in app/etc/config.php.';
+        }
+
+        foreach (['minify_files', 'enable_js_bundling', 'merge_files'] as $configKey) {
```

Comment:
Are these options 'minify_files', 'enable_js_bundling', 'merge_files' are enabled by default in Magento?
If they aren't enabled by default, next check `!isset($jsConfig[$configKey])` will mean that config is disabled and it won't be an error.

Reply:
I believe these values are `'0'` when the config file is dumped by default, however part of the intent is that the user should _know_ these options are being disabled, not just rely on the defaults from Magento, so that we ensure future compatibility with Baler